readthedocs/proxito/views/mixins.py

```diff
@@ -96,31 +101,26 @@ def _is_audit_enabled(self, project):
         """
         return False
 
-    def _serve_docs_python(self, request, final_project, path):
-        """
-        Serve docs from Python.
-
-        .. warning:: Don't do this in production!
-        """
-        log.debug('Django serve.', path=path, project_slug=final_project.slug)
-
-        root_path = build_media_storage.path('')
-        # Serve from Python
-        return serve(request, path, root_path)
+    def _serve_static_file(self, request, path):
+        if settings.PYTHON_MEDIA:
+            return self._serve_file_from_python(request, path, staticfiles_storage)
+        return self._serve_file_from_nginx(path, root_path="proxito-static")
 
-    def _serve_docs_nginx(self, request, final_project, version_slug, path, download):
+    def _serve_file_from_nginx(self, path, root_path):
         """
-        Serve docs from nginx.
+        Serve a file from nginx.
 
         Returns a response with ``X-Accel-Redirect``, which will cause nginx to
         serve it directly as an internal redirect.
-        """
 
+        :param path: The path of the file to serve.
+        :param root_path: The root path of the internal redirect.
+        """
         original_path = copy.copy(path)
-        if not path.startswith('/proxito/'):
+        if not path.startswith(f"/{root_path}/"):
             if path[0] == '/':
                 path = path[1:]
-            path = f'/proxito/{path}'
+            path = f"/{root_path}/{path}"
```

Comment:
I think it would be safer if we always add the root_path to the path, I had to use `proxito-static` instead of `static` for the internal redirect because locally the storage path is `/static/foo.js`, so we want the final path to be `/static/static/foo.js` but adding the extra `static` was being skipped here.